client/app/scripts/superdesk-publish/filters/publish-filters.js

```diff
@@ -483,9 +490,141 @@ function ProductionTestController($scope, filters, notify, $location, $window) {
     };
 }
 
+FilterSearchController.$inject = ['$scope', 'filters', 'notify'];
+function FilterSearchController($scope, filters, notify) {
+    $scope.filterCondition = null;
+    $scope.operatorLookup = {};
+    $scope.valueLookup = {};
+    $scope.valueFieldLookup = {};
+    $scope.searchResult = null;
+    $scope.publishFiltersLookup = {};
+
+    populateData();
+    fetchPublishFilters();
+    $scope.isListValue = function() {
+        if ($scope.filterCondition != null) {
+            return _.contains(['in', 'nin'], $scope.filterCondition.operator) && $scope.valueLookup[$scope.filterCondition.field];
+        }
+    };
+
+    $scope.hideList = function () {
+        return true;
+    };
+
+    $scope.handleKey = function(event) {
+        if ($scope.filterCondition.values.length > 0) {
+            notify.error(gettext('single value is required'));
+            event.preventDefault();
+            event.stopPropagation();
+            return false;
+        }
+    };
+
+    $scope.resetValues = function() {
+        $scope.searchResult = null;
+        $scope.filterCondition.values.length = 0;
```

Comment:
Just curious - why do we need a new (empty) Array here, isn't emptying the existing one enough?

(also, if there perhaps exist any other references to this array, modifying the existing Array in place is probably what we want to do IMO)